pkg/domain/infra/abi/play.go

```diff
@@ -182,20 +182,22 @@ func (ic *ContainerEngine) playKubePod(ctx context.Context, podName string, podY
 		return nil, err
 	}
 	if options.Network != "" {
-		switch strings.ToLower(options.Network) {
-		case "bridge", "host":
+		ns, cniNets, netOpts, err := specgen.ParseNetworkString(options.Network)
+		if err != nil {
+			return nil, err
+		}
+
+		if (ns.IsBridge() && len(cniNets) == 0) || ns.IsHost() {
 			return nil, errors.Errorf("invalid value passed to --network: bridge or host networking must be configured in YAML")
-		case "":
-			return nil, errors.Errorf("invalid value passed to --network: must provide a comma-separated list of CNI networks")
-		default:
-			// We'll assume this is a comma-separated list of CNI
-			// networks.
-			networks := strings.Split(options.Network, ",")
-			logrus.Debugf("Pod joining CNI networks: %v", networks)
-			p.NetNS.NSMode = specgen.Bridge
-			p.CNINetworks = append(p.CNINetworks, networks...)
+		}
+		logrus.Debugf("Pod joining CNI networks: %v", cniNets)
```

Comment:
This message should include Pod id if tunneled code comes via this code path.

Reply:
Added podName to output.